tools/cmake: update to 3.31.2

zheshifandian opened the pull request:
Release notes: https://cmake.org/cmake/help/latest/release/3.31.html

Refresh patches:
- 120-curl-fix-libressl-linking.patch
- 130-bootstrap_parallel_make_flag.patch
- 150-zstd-libarchive.patch
- 160-disable_xcode_generator.patch

Neustradamus commented:
Replaces:
- https://github.com/openwrt/openwrt/pull/17049

Neustradamus commented:
Replaces:
- 3.31.0: https://github.com/openwrt/openwrt/pull/16884
- 3.31.1: https://github.com/openwrt/openwrt/pull/17049

hauke commented:
Please refresh the patches:
```
make tools/cmake/{check,refresh} FIXUP=1
```

zheshifandian commented:
> Please refresh the patches:
> 
> ```
> make tools/cmake/{check,refresh} FIXUP=1
> ```

done

zheshifandian committed:
tools/cmake: update to 3.31.2
Release notes: https://cmake.org/cmake/help/latest/release/3.31.html

Refresh patches:
- 120-curl-fix-libressl-linking.patch
- 130-bootstrap_parallel_make_flag.patch
- 150-zstd-libarchive.patch
- 160-disable_xcode_generator.patch

Signed-off-by: Jack Sun <sunjiazheng321521@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17056
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
